src/state/pages/selectors.js

```diff
@@ -92,14 +92,23 @@ export const getPageTreePages = createSelector(
   (pages, pageChildren, pagesStatus, pagesTitles, locale) => (
     getPagesOrder(pageChildren)
       .filter(pageCode => isVisible(pageCode, pages, pagesStatus))
-      .map(pageCode => ({
-        ...pages[pageCode],
-        ...PAGE_STATUS_DEFAULTS,
-        ...pagesStatus[pageCode],
-        depth: getDepth(pages, pageCode),
-        isEmpty: !(pageChildren[pageCode] && pageChildren[pageCode].length),
-        title: pagesTitles[pageCode][locale],
-      }))),
+      .map((pageCode) => {
+        const isEmpty = !(pageChildren[pageCode] && pageChildren[pageCode].length);
+        let pagePublish = false;
```

Comment:
I think "pagePublish" does not reflect the nature of this property (indicating if the page has children which are published)
Could you rename it as "hasPublishedChildren" ?